### DISCUSSION TITLE
Fatal Error during setup of Zephyr

### DISCUSSION THREAD

**Discussion by hafizmqasim** [zephyrproject-rtos/zephyr#67654] [Q&A] [answered]
Hi All, 

I am following the page to setup Zephyr https://docs.zephyrproject.org/latest/develop/getting_started/index.html

I am facing following error 

(.venv) C:\Users\m3n1iz>west init zephyrproject
FATAL ERROR: already initialized in C:\Users\m3n1iz, aborting.
  Hint: if you do not want a workspace there,
  remove this directory and re-run this command:

  C:\Users\m3n1iz\.west

Here is the list of commands that are executed step by step. 
`C:\Windows\System32>**choco feature enable -n allowGlobalConfirmation**
Chocolatey v2.2.2
Nothing to change. Config already set.

C:\Windows\System32>**choco install cmake --installargs 'ADD_CMAKE_TO_PATH=System'**
Chocolatey v2.2.2
Installing the following packages:
cmake
By installing, you accept licenses for the packages.
cmake v3.27.9 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.

Chocolatey installed 0/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Warnings:
 - cmake - cmake v3.27.9 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.

C:\Windows\System32>**choco install ninja gperf python311 git dtc-msys2 wget 7zip**
Chocolatey v2.2.2
Installing the following packages:
ninja;gperf;python311;git;dtc-msys2;wget;7zip
By installing, you accept licenses for the packages.
ninja v1.11.1 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
gperf v3.1.0 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
python311 v3.11.6 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
git v2.43.0 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
dtc-msys2 v1.5.1 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
Wget v1.21.4 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
7zip v23.1.0 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.

Chocolatey installed 0/7 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Warnings:
 - 7zip - 7zip v23.1.0 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
 - dtc-msys2 - dtc-msys2 v1.5.1 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
 - git - git v2.43.0 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
 - gperf - gperf v3.1.0 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
 - ninja - ninja v1.11.1 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
 - python311 - python311 v3.11.6 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.
 - Wget - Wget v1.21.4 already installed.
 Use --force to reinstall, specify a version to install, or try upgrade.

C:\Windows\System32>**cd %HOMEPATH%**

C:\Users\m3n1iz>**python -m venv zephyrproject\.venv**

C:\Users\m3n1iz>**zephyrproject\.venv\Scripts\activate.bat**

(.venv) C:\Users\m3n1iz>**pip install west**
Requirement already satisfied: west in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (1.2.0)
Requirement already satisfied: colorama in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from west) (0.4.6)
Requirement already satisfied: PyYAML>=5.1 in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from west) (6.0.1)
Requirement already satisfied: pykwalify in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from west) (1.8.0)
Requirement already satisfied: setuptools in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from west) (65.5.0)
Requirement already satisfied: packaging in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from west) (23.2)
Requirement already satisfied: docopt>=0.6.2 in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from pykwalify->west) (0.6.2)
Requirement already satisfied: python-dateutil>=2.8.0 in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from pykwalify->west) (2.8.2)
Requirement already satisfied: ruamel.yaml>=0.16.0 in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from pykwalify->west) (0.18.5)
Requirement already satisfied: six>=1.5 in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from python-dateutil>=2.8.0->pykwalify->west) (1.16.0)
Requirement already satisfied: ruamel.yaml.clib>=0.2.7 in c:\users\m3n1iz\zephyrproject\.venv\lib\site-packages (from ruamel.yaml>=0.16.0->pykwalify->west) (0.2.8)

(.venv) C:\Users\m3n1iz>**west init zephyrproject**
FATAL ERROR: already initialized in C:\Users\m3n1iz, aborting.
  Hint: if you do not want a workspace there,
  remove this directory and re-run this command:

  C:\Users\m3n1iz\.west

(.venv) C:\Users\m3n1iz>`

**Comment by hafizmqasim** [comment]
Removed .west folder present in "C:\Users\m3n1iz\" and executed the following command again. 
west init zephyrproject

But now got the following error. 

(.venv) C:\Users\m3n1iz>west init zephyrproject
=== Initializing in C:\Users\m3n1iz\zephyrproject
--- Cloning manifest repository from https://github.com/zephyrproject-rtos/zephyr
Cloning into 'C:\Users\m3n1iz\zephyrproject\.west\manifest-tmp'...
remote: Enumerating objects: 974419, done.
remote: Counting objects: 100% (28340/28340), done.
remote: Compressing objects: 100% (1541/1541), done.
error: 4384 bytes of body are still expected43 MiB | 361.00 KiB/s
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
FATAL ERROR: command exited with status 128: git clone https://github.com/zephyrproject-rtos/zephyr 'C:\Users\m3n1iz\zephyrproject\.west\manifest-tmp'

**Comment by hafizmqasim** [comment]
Again removed .west folder present in "C:\Users\m3n1iz\zephyrproject" and executed the following command again. 

west init zephyrproject

will post the result once completed.

**Reply by ndrs-pst** [reply]
@hafizmqasim, I followed your instructions and successfully ran `west init zephyrproject` within [Windows Sandbox](https://learn.microsoft.com/en-us/windows/security/application-security/application-isolation/windows-sandbox/windows-sandbox-overview) without encountering any errors.
I recommend trying it first with Windows Sandbox.

**Reply by hafizmqasim** [reply]
Removing .west folder does not help. Facing same error. It seems to be an issue with git command working without compression. Probably, we need to update project.py but not sure.

**Reply by hafizmqasim** [reply, marked as answer]
Fixed the problem. 

Updated the project.py file located at "C:\Users\m3n1iz\zephyrproject\.venv\Lib\site-packages\west\app" as 

self.check_call(['git', 'clone'**, '--depth=1'**] + branch_opt +
                            [manifest_url, os.fspath(tempdir)])

By setting the depth=1 in git clone command, west init zephyrproject is successfully completed. 

=== Initializing in C:\Users\m3n1iz\zephyrproject
--- Cloning manifest repository from https://github.com/zephyrproject-rtos/zephyr
Cloning into 'C:\Users\m3n1iz\zephyrproject\.west\manifest-tmp'...
remote: Enumerating objects: 36447, done.
remote: Counting objects: 100% (36447/36447), done.
remote: Compressing objects: 100% (28225/28225), done.
remote: Total 36447 (delta 8453), reused 19410 (delta 6164), pack-reused 0
Receiving objects: 100% (36447/36447), 71.37 MiB | 1003.00 KiB/s, done.
Resolving deltas: 100% (8453/8453), done.
Updating files: 100% (31631/31631), done.
--- setting manifest.path to zephyr
=== Initialized. Now run "west update" inside C:\Users\m3n1iz\zephyrproject.
